### ISSUE TITLE
Enable Java lints for the plugins/packages repos

### ISSUE THREAD

**Issue by bparrishMines** [flutter/flutter#119836]
A linter runs for the Dart code and Objective-C/Swift code in the [flutter/plugins](https://github.com/flutter/plugins) and [flutter/packages](https://github.com/flutter/packages) repos. 

This issue tracks enabling it for Java and Kotlin.

We should be able to just use the command provided by gradle:
See https://developer.android.com/studio/write/lint

```
gradlew lint
./gradlew lint
```

I would imagine we would initially need a config file to exclude plugins and fix the lint warnings incrementally. 

cc @camsim99 @reidbaker @stuartmorgan FYI

**Comment by bparrishMines** [comment]
@stuartmorgan Is this something we could do before or after https://github.com/flutter/flutter/issues/113764?

**Comment by stuartmorgan** [comment]
I we do it before, we just need to make sure it's added in parallel to both repos so we don't re-diverge their test set.

> I would imagine we would initially need a config file to exclude plugins and fix the lint warnings incrementally.

We already have lint_baseline.xml set up as part of uploading build warning output, so hopefully we could just update that.

**Comment by stuartmorgan** [comment]
Would this address https://github.com/flutter/flutter/issues/91868 for Android, or are lint warnings and build warnings disjoint?

**Comment by reidbaker** [comment]
I am pro making this happen.

**Comment by stuartmorgan** [comment]
> We already have lint_baseline.xml set up as part of uploading build warning output

... which we have because I set up a command over a year ago that runs this lint command. I forgot that's what it was doing 🤦🏻

Closing as a duplicate of https://github.com/flutter/flutter/issues/87071. I'll look into why some warnings aren't being flagged by this, like https://github.com/flutter/packages/pull/3273. Maybe it's just the exclusions, and what we need is a push to remove exclusions that were set up when we turned this on?

**Comment by stuartmorgan** [comment]
Also, while not relevant to it not failing CI, it appears that running the lint command twice only outputs warnings the first time, which is definitely not a behavior we want.

**Comment by stuartmorgan** [comment]
> It looks like having warnings output doesn't cause it to return a non-zero exit code. Re-opening while I investigate further.

Investigating further: I'm not crazy thinking I had to set up baselines to get things to pass when I first did this, I just wasn't looking at the output closely enough. The output I was seeing was:

```
> Task :camera_android:lintDebug
Lint found no errors or warnings

2 errors/warnings were listed in the baseline file (/Users/stuartmorgan/src/packages/packages/camera/camera_android/android/lint-baseline.xml) but not found in the project; perhaps they have been fixed?

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/7.0.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
52 actionable tasks: 52 executed
/Users/stuartmorgan/src/packages/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java:158: warning: [deprecation] get(int,int) in CamcorderProfile has been deprecated
          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_HIGH);
                                 ^
/Users/stuartmorgan/src/packages/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java:162: warning: [deprecation] get(int,int) in CamcorderProfile has been deprecated
          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_2160P);
                                 ^
/Users/stuartmorgan/src/packages/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java:166: warning: [deprecation] get(int,int) in CamcorderProfile has been deprecated
          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_1080P);
                                 ^
/Users/stuartmorgan/src/packages/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java:170: warning: [deprecation] get(int,int) in CamcorderProfile has been deprecated
          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_720P);
                                 ^
/Users/stuartmorgan/src/packages/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java:174: warning: [deprecation] get(int,int) in CamcorderProfile has been deprecated
          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_480P);
                                 ^
/Users/stuartmorgan/src/packages/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java:178: warning: [deprecation] get(int,int) in CamcorderProfile has been deprecated
          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_QVGA);
                                 ^
/Users/stuartmorgan/src/packages/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java:182: warning: [deprecation] get(int,int) in CamcorderProfile has been deprecated
          return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_LOW);
                                 ^
7 warnings
```

I missed the `Lint found no errors or warnings` bit. So those warnings are being dumped to the console when I run lint as part of the overall gradle run, but aren't part of the lint step itself. That may also explain why it doesn't output anything when run a second time; it's not the step actually being targetted.

**Comment by stuartmorgan** [comment]
Okay, it looks like what's happening here is that the Gradle `lint` is different from `javac` lints, and the deprecation warnings are coming from the latter. So this is a duplicate, and the above would be https://github.com/flutter/flutter/issues/91868
